Delete old test results prior to running tests or computing coverage.

rjust opened the pull request:
Some additional clean up (variable naming and making the file name for all_tests a parameter) would be great, but should go into a separate PR.

Greg4cr commented:
Looks good to me.

jose commented:
Looks good to me.

Just a minor comment.  How about something more `perl` and less `system`? :-)
E.g.:

```perl
open my $fh, '>', "$fail_tests" or die $!;
close $fh;
```

or

```perl
truncate "$fail_tests", 0 or die $!;
```

rjust commented:
@jose, good point. I replaced the `system` code with standard `unlink`. Also, I refactored the common code into a util subroutine.

Let me know what you think.
